job-runner/chain/reaper_test.go

```diff
@@ -1037,6 +1357,136 @@ func TestSuspendedReaper(t *testing.T) {
 	}
 }
 
+// test suspendedChainReaper.Run
+func TestSuspendedReaperUnknownState(t *testing.T) {
+	// Job Chain:
+	//          6
+	//         /
+	//       2 - 5
+	//     /  \
+	// -> 1    4
+	//     \  /
+	//      3
+	// jobs 3, 5, 6 running
+
+	reqId := "test_suspended_reaper"
+	jc := &proto.JobChain{
+		RequestId: reqId,
+		Jobs:      testutil.InitJobs(6),
+		AdjacencyList: map[string][]string{
+			"job1": {"job2", "job3"},
+			"job2": {"job4", "job5", "job6"},
+			"job3": {"job4"},
+		},
+	}
+
+	// Make job6 a separate sequence
+	job := jc.Jobs["job6"]
+	job.SequenceId = "job6"
+	job.SequenceRetry = 1
```

Comment:
eg. related to my comment on FailedJobs, if you did a test like this but without giving job6 any retries, what you'd expect is for the request to fail, but as is I don't think it would.